platform/ios/src/MGLMapView.mm

```diff
@@ -1717,14 +1717,16 @@ - (void)handleRotateGesture:(UIRotationGestureRecognizer *)rotate
 {
     if ( ! self.isRotateEnabled) return;
 
-    [self cancelTransitions];
-
     if (MGLDegreesFromRadians(self.currentRotation) < self.rotationThreshold && self.isZooming) {
+        [self notifyGestureDidBegin];
```

Comment:
Is this when the gesture begins? This looks like where the gesture is being explicit kept from beginning.

Reply:
The exception that I see from the `return;` is happening when `_changeDelimiterSuppressionDepth` is < 0. Just calling `_changeDelimiterSuppressionDepth++;` makes the first rotation jump to a new rotation. I still see that behavior occasionally by calling `notifyGestureDidBegin`.

Right now, this code is still being called when the rotation gesture ends/fails, even if the map doesn't rotate: 
https://github.com/mapbox/mapbox-gl-native/blob/4474ef6340bd92d1358adc393e7c9f88871f4afb/platform/ios/src/MGLMapView.mm#L1808-L1815

Removing `return` may work better. The camera animations, tracking, etc. are going to be cancelled by the pinch gesture recognizer that is also being triggered.